### ISSUE TITLE
Extension does not work

### ISSUE THREAD

**Issue by tsuliwaensis** [theRealPadster/spicetify-hide-podcasts#133]
### 🔍 Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### ℹ Environment / Computer Info

```Markdown
- Spotify version: 1.2.32.997.g4c6498b6 [macOS]
- Spicetify version: 2.33.2
```

### 📝 Description

The extension appears in the profile menu but nothing happens when you try to enable it. A checkbox doesn't appear next to "Enabled" and Podcasts are still appearing.

**Comment by theRealPadster** [comment]
Are you running the latest version of the extension? I'd suggest installing it locally (remove from Marketplace and install manually) to be sure. The latest version of Spicetify fixed a bug where it wasn't showing checkmarks in menu items, though they still functioned. Now it looks like it shows the checkmark when you click it, but not on first load of Spotify. If you go to the search page, and toggle the extension on and off a few times, it should hide/show a few of the genre cards. Can you test this?

**Comment by theRealPadster** [comment]
Fix for initial menu item state is in this commit. Will need to wait for next spicetify release for that. 

https://github.com/spicetify/spicetify-cli/pull/2905

**Comment by tsuliwaensis** [comment]
I reinstalled Spicetify (version 2.35.1) and Marketplace and the extension sort of works. I don't see the Podcast in the home page anymore, but the buttons still appear on the home page and the sidebar.

**Comment by theRealPadster** [comment]
> I reinstalled Spicetify (version 2.35.1) and Marketplace and the extension sort of works. I don't see the Podcast in the home page anymore, but the buttons still appear on the home page and the sidebar.

Can you post a screenshot? I'm not sure what buttons you're referring to.

**Comment by tsuliwaensis** [comment]
I have a theme applied and that might be the reason why it's not being hidden

![2024-03-22_08-44-20](https://github.com/theRealPadster/spicetify-hide-podcasts/assets/40817465/6f61c015-15dc-4aac-93be-686d462f29a9)

**Comment by theRealPadster** [comment]
> I have a theme applied and that might be the reason why it's not being hidden
> 
> ![2024-03-22_08-44-20](https://private-user-images.githubusercontent.com/40817465/316042465-6f61c015-15dc-4aac-93be-686d462f29a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExMTkwNTMsIm5iZiI6MTcxMTExODc1MywicGF0aCI6Ii80MDgxNzQ2NS8zMTYwNDI0NjUtNmY2MWMwMTUtMTVkYy00YWFjLTkzYmUtNjg2ZDQ2MmYyOWE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIyVDE0NDU1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2NTMzNjUzYjUzNzdkZmRkNWY4MWViMTczZDUwNDI0MjY4YzljZjgyNmVkNTIyZjhjMzJmZjM1OTdjOGVhMjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zJ_BsYUBQkl-rLxqZfN9Dc8Q8z840826_Ej4MozYQ3U)

Is the rectangle part this UI element? What theme are you using? Can you post a bigger screenshot?
![image](https://github.com/theRealPadster/spicetify-hide-podcasts/assets/1565883/df46ab26-a8e1-48db-aa30-0e01262a639d)

**Comment by theRealPadster** [comment]
It looks like I just don't have that UI on my Spotify for some reason. 

Someone on discord shared the UI element, so I might try and add something to block it when I have a chance (in the next few days) but hard to test without having it on my Spotify. 
```html
<button role="checkbox" aria-checked="false" aria-label="Podcasts" data-encore-id="chip" class="Chip__ChipComponent-sc-ry3uox-0 ChipComponent-checkbox-chip-useBrowserDefaultFocusStyle"><span class="ChipInner__ChipInnerComponent-sc-1ly6j4j-0 ChipInnerComponent">Podcasts</span></button>
```

**Comment by tsuliwaensis** [comment]
I removed the theme and it's still there. I'm on a Mac, if that makes a difference. Here's a fullscreen screenshot

![2024-03-24_08-59-35](https://github.com/theRealPadster/spicetify-hide-podcasts/assets/40817465/db5e043e-dd41-4109-8c95-6645f507d58f)

**Comment by theRealPadster** [comment]
Yeah I still don't have this UI. Marking as blocked for now.

**Comment by 80H-d** [comment]
![image](https://github.com/theRealPadster/spicetify-hide-podcasts/assets/145611979/ea84c3fe-d5ed-4f8c-9713-739a39a2dbc4)
/shrug

**Comment by theRealPadster** [comment]
> ![image](https://private-user-images.githubusercontent.com/145611979/346028486-ea84c3fe-d5ed-4f8c-9713-739a39a2dbc4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAyNzc3MTQsIm5iZiI6MTcyMDI3NzQxNCwicGF0aCI6Ii8xNDU2MTE5NzkvMzQ2MDI4NDg2LWVhODRjM2ZlLWQ1ZWQtNGY4Yy05NzEzLTczOWEzOWEyZGJjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNlQxNDUwMTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzRlN2Y0Y2RhZjU2ZTQ3ZWFiNmNlODNkN2ZlMjVlMzNhN2JhYjAxMTQxYTJiODc1Yjc0YWI1MDQ5ZGEwYzE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._uf7h5H7Za8CJFljSXNl6JBGyW9L9e6YbCto1lJbXPE) on windows 11 /shrug

Yeah I've got that UI on my W11 now. I don't have podcasts or audiobooks there though, I'd guess because I've never interacted with them before. I looked at the way the ones that I do have are set up, and got it working to hide those ones, then adjusted it to be for podcasts + audiobooks. That's been updated in the latest `hidePodcasts.js` file. Could you give it a try?

**Comment by theRealPadster** [comment]
Closing as inactive.
